### PR TITLE
[Costmanagement] Add CostManagement Generated Code and Tests

### PR DESCRIPTION
**Pull request by Youri970410**
## Description
Generate CostManagement code base on Version '2020-06-01' and test all cmdlets except one Syntax of 'Invoke-AzCostManagementQuery'.

## Checklist

- [ ] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [ ] The title of the PR is clear and informative
- [ ] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [ ] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
        - {Please put the link here}
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

File: src/CostManagement/custom/Invoke-AzCostManagementQuery.ps1 [hunk 1]
```diff
@@ -0,0 +1,342 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ----------------------------------------------------------------------------------
+
+<#
+.Synopsis
+Query the usage data for scope defined.
+.Description
+Query the usage data for scope defined.
+.Example
```
Comment (dolauli) [inline src/CostManagement/custom/Invoke-AzCostManagementQuery.ps1:21]: Remove empty examples

File: src/CostManagement/docs/Invoke-AzCostManagementExecuteExport.md [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+## DESCRIPTION
+The operation to execute an export.
+
+## EXAMPLES
+
+### Example 1: Invoke Export by ExportName and Scope
+```powershell
+PS C:\> Invoke-AzCostManagementExecuteExport -ExportName 'TestExport' -Scope 'subscriptions/**********'
+
+{{ Add output here }}
```
Comment (dolauli) [inline src/CostManagement/docs/Invoke-AzCostManagementExecuteExport.md:36]: Clean code like these

File: src/CostManagement/docs/Invoke-AzCostManagementQuery.md [hunk 3]
```diff
@@ -0,0 +1,330 @@
...
+ [-DatasetGranularity <GranularityType>] [-DatasetGrouping <IQueryGrouping[]>] [-TimePeriodFrom <DateTime>]
+ [-TimePeriodTo <DateTime>] [-DefaultProfile <PSObject>] [-Confirm] [-WhatIf] [<CommonParameters>]
+```
+
+## DESCRIPTION
+Query the usage data for scope defined.
+
+## EXAMPLES
+
+### Example 1: {{ Add title here }}
```
Comment (dolauli) [inline src/CostManagement/docs/Invoke-AzCostManagementQuery.md:37]: Please add the title

File: src/CostManagement/custom/New-AzCostManagementQueryColumnObject.ps1 [hunk 4]
```diff
@@ -0,0 +1,48 @@
+
```
Comment (dolauli) [inline src/CostManagement/custom/New-AzCostManagementQueryColumnObject.ps1:1]: Looks this client side cmdlet is not required.

File: src/CostManagement/readme.md [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+> `autorest`
+---
+### AutoRest Configuration
+> see https://aka.ms/autorest
+
+``` yaml
+require:
+  - $(this-folder)/../readme.azure.noprofile.md
+input-file:
+  - $(repo)/specification/cost-management/resource-manager/Microsoft.CostManagement/stable/2020-06-01/costmanagement.json
```
Comment (dolauli) [inline src/CostManagement/readme.md:51]: Please specify swagger through commit id instead of using repo

**Review by dolauli — CHANGES_REQUESTED** [review]
Some changes are required

File: src/CostManagement/custom/New-AzCostManagementQueryColumnObject.ps1 (hunk 4 above)
Comment (Youri970410) [reply src/CostManagement/custom/New-AzCostManagementQueryColumnObject.ps1:1]: Yes.This cmdlet will be deleted in subsequent submissions.

File: src/CostManagement/readme.md (hunk 5 above)
Comment (Youri970410) [reply src/CostManagement/readme.md:51]: Got it.thanks.

File: src/CostManagement/custom/New-AzCostManagementExport.ps1 [hunk 6]
```diff
@@ -0,0 +1,244 @@
...
+The operation to create or update a export.
+Update operation requires latest eTag to be set in the request.
+You may obtain the latest eTag by performing a get operation.
+Create operation does not require eTag.
+.Description
+The operation to create or update a export.
+Update operation requires latest eTag to be set in the request.
+You may obtain the latest eTag by performing a get operation.
+Create operation does not require eTag.
+.Example
```
Comment (dolauli) [inline src/CostManagement/custom/New-AzCostManagementExport.ps1:27]: Remove examples, outputs and link from the comments.

File: src/CostManagement/custom/Invoke-AzCostManagementQuery.ps1 [hunk 7]
```diff
@@ -0,0 +1,334 @@
...
+      Value <String[]>: Array of values to use for comparison
+    [Not <IQueryFilter>]: The logical "NOT" expression.
+    [Or <IQueryFilter[]>]: The logical "OR" expression. Must have at least 2 items.
+    [Tag <IQueryComparisonExpression>]: Has comparison expression for a tag
+  [DatasetGranularity <GranularityType?>]: The granularity of rows in the query.
+  [DatasetGrouping <IQueryGrouping[]>]: Array of group by expression to use in the query. Query can have up to 2 group by clauses.
+    Name <String>: The name of the column to group.
+    Type <QueryColumnType>: Has type of the column to group.
+.Link
+https://docs.microsoft.com/en-us/powershell/module/az.cost/invoke-azcostmanagementusagequery
```
Comment (dolauli) [inline src/CostManagement/custom/Invoke-AzCostManagementQuery.ps1:78]: Link is incorrect here. should not az.cost. Just remove the link part.

File: src/CostManagement/docs/Update-AzCostManagementExport.md [hunk 8]
```diff
@@ -0,0 +1,421 @@
...
+
+## DESCRIPTION
+The operation to create or update a export.
+Update operation requires latest eTag to be set in the request.
+You may obtain the latest eTag by performing a get operation.
+Create operation does not require eTag.
+
+## EXAMPLES
+
+### Example 1: {{ Add title here }}
```
Comment (dolauli) [inline src/CostManagement/docs/Update-AzCostManagementExport.md:46]: Please add examples here.

File: src/CostManagement/examples/Update-AzCostManagementExport.md [hunk 9]
```diff
@@ -0,0 +1,18 @@
+### Example 1: {{ Add title here }}
+```powershell
+PS C:\> {{ Add code here }}
```
Comment (dolauli) [inline src/CostManagement/examples/Update-AzCostManagementExport.md:3]: Missing examples

**Review by dolauli — DISMISSED** [review]
1. some examples are missing
2. In customized ps1 scripts, examples and link are not needed.

**Review by wyunchi-ms — APPROVED** [review]

File: src/CostManagement/examples/Update-AzCostManagementExport.md (hunk 9 above)
Comment (Youri970410) [reply src/CostManagement/examples/Update-AzCostManagementExport.md]: Got it,thanks.

File: src/CostManagement/docs/Update-AzCostManagementExport.md (hunk 8 above)
Comment (Youri970410) [reply src/CostManagement/docs/Update-AzCostManagementExport.md]: Got it,thanks.

**Review by wyunchi-ms — APPROVED** [review]

**Pull request merged**